frontend/test/metabase/visualizations/visualizations/Table.unit.spec.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import { thaw } from "icepick";
+import userEvent from "@testing-library/user-event";
+import { renderWithProviders, screen, within } from "__support__/ui";
+
+import {
+  SAMPLE_DATABASE,
+  ORDERS,
+  metadata,
+} from "__support__/sample_database_fixture";
+import ChartSettings from "metabase/visualizations/components/ChartSettings";
+
+import Question from "metabase-lib/Question";
+
+const setup = () => {
+  const Container = () => {
+    const [question, setQuestion] = useState(
+      new Question(
+        {
+          dataset_query: {
+            type: "query",
+            query: {
+              "source-table": ORDERS.id,
+            },
+          },
+          database: SAMPLE_DATABASE.id,
+          visualization_settings: {},
+        },
+        metadata,
+      ),
+    );
+
+    const onChange = update => {
+      setQuestion(q => {
+        const newQuestion = q.updateSettings(update);
+        return new Question(thaw(newQuestion.card()), metadata);
```

Comment:
Need to thaw the card since we freeze it when calling updateSettings (we call `icepick.assoc` and write `visualization_settings`). This causes the ChartSettings component to start throwing errors when it re-renders.